sources/pmwebapi.go

```diff
@@ -0,0 +1,164 @@
+package sources
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type GetToken struct {
```

Comment:
You will probably want to change this to `getToken` since it is only used in this file. Otherwise, it will be exported unnecessarily and might cause issues in the future. Note that you will have to change these in the functions below where you use them. Also, it is entirely possible that a few of these need to be kept uppercase since they are embedded structs, but I am not positive on this.